src/aap_eda/conf/registry.py

```diff
@@ -0,0 +1,145 @@
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from collections import OrderedDict
+from dataclasses import dataclass
+from typing import Any
+
+import yaml
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from aap_eda.core import models
+from aap_eda.core.utils.crypto.base import SecretValue
+
+ENCRYPTED_STRING = "$encrypted$"
+
+
+@dataclass
+class RegistryData(object):
+    name: str
+    default: Any
+    type: type = str
+    read_only: bool = False
+    encrypted: bool = False
+
+
+_APPLICATION_SETTING_REGISTRIES = [
+    RegistryData(
+        name="INSIGHTS_TRACKING_STATE",
+        type=bool,
+        default=False,
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_URL",
+        read_only=True,
+        default=settings.AUTOMATION_ANALYTICS_URL,
+    ),
+    RegistryData(
+        name="REDHAT_USERNAME",
+        default="",
+    ),
+    RegistryData(
+        name="REDHAT_PASSWORD",
+        encrypted=True,
+        default="",
+    ),
+    RegistryData(
+        name="SUBSCRIPTIONS_USERNAME",
+        default="",
+    ),
+    RegistryData(
+        name="SUBSCRIPTIONS_PASSWORD",
+        encrypted=True,
+        default="",
+    ),
+    RegistryData(
+        name="INSIGHTS_CERT_PATH",
+        read_only=True,
+        default=settings.INSIGHTS_CERT_PATH,
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_LAST_GATHER",
+        default="",
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_LAST_ENTRIES",
+        type=dict,
+        default={},
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_GATHER_INTERVAL",
+        type=int,
+        default=14400,
+    ),
+]
+
+
+class SettingsRegistry(object):
+    def __init__(self):
+        self._registry = OrderedDict()
+        for registry_data in _APPLICATION_SETTING_REGISTRIES:
+            self.register(registry_data)
+
+    def register(self, registry_data: RegistryData) -> None:
+        if registry_data.name in self._registry:
+            raise ImproperlyConfigured(
+                f"Setting {registry_data.name} is already registered."
+            )
+        self._registry[registry_data.name] = registry_data
+        if registry_data.read_only:
+            update_method = models.Setting.objects.update_or_create
+        else:
+            update_method = models.Setting.objects.get_or_create
+        update_method(
+            key=registry_data.name, defaults={"value": registry_data.default}
+        )
+
+    def unregister(self, setting: str) -> None:
```

Comment:
This is method is not used. What would be its purpose? I don't see why we would want to do that.

Reply:
again. it is copied. I will remove